iep-004/README.adoc

```diff
@@ -0,0 +1,260 @@
+
+This specification only outlines the architecture for the "Public Production"
+footnoteref:[iep2,https://github.com/jenkins-infra/iep/tree/master/iep-002]
+Kubernetes cluster. It is expected that the project will run multiple
+Kubernetes clusters, following this architecture, depending on the access
+control requirements for each discrete Kubernetes.
+
+At a high level Kubernetes is a master/agent architecture which would live in a
+*single region*. For the purposes of Jenkins infrastructure, the production
+Kuberenetes clusters would be located in the "East US" Azure region.
+footnoteref:[regions,https://azure.microsoft.com/en-us/regions/]
+
+The virtual machines are all
+link:https://azure.microsoft.com/en-us/pricing/details/virtual-machines/series/#d-series[D-series]
+instances in order to provide an ideal balance of cost and performance. For
+more, see <<Costs>> below.
+
+
+*Kubernetes master*
+
+A cluster would run a single D3v2 instance as the master node
+
+
+*Kubernetes agents*
+
+A cluster would run agents using a
+link:https://azure.microsoft.com/en-us/services/virtual-machine-scale-sets/[Scale Set]
+of D2v2 instances, with a minimum of three agents running at all times.
+
+
+
+[NOTE]
+====
+Following
+link:https://github.com/jenkins-infra/iep/tree/master/iep-003[IEP-3]
+the <<reference-implementation>> uses Terraform to describe the infrastructure
+necessary to support this specification.
+====
+
+
+=== Monitoring
+
+As the Jenkins project already uses
+link:http://datadoghq.com[Datadog]
+for monitoring our production infrastructure, the Kubernetes cluster must
+integrate appropriately.
+
+This will be accomplished using the
+link:http://docs.datadoghq.com/integrations/kubernetes/[Datadog/Kubernetes integration]
+maintained by Datadog themselves.
+
+The integration is will provide metrics for both:
+
+* Kubernetes cluster health/status
+* Container health/status running atop the cluster
+
+The metrics required are not specified here under the assumption that all
+metrics appropriate for ensuring stable service levels of project
+infrastructure will be collected.
+
+=== Logging
+
+[NOTE]
+====
+This section is still a work in progress, awaiting information to be provided
+by
+link:https://github.com/olblak[Olivier Vernin]
+====
+
+Centralized logging for applications hosted within Kubernetes will be provided
+a combination of containers in the cluster running
+link:https://en.wikipedia.org/wiki/Fluentd[Fluentd]
+and the Microsoft
+link:http://www.microsoft.com/en-us/cloud-platform/operations-management-suite[Operations Management Suite]
+(OMS).
+
+.Data flow for k8s logs
+[source]
+----
++-------------------------+
+|  Kubernetes             |
+| |          ^            |
+| | +-----+  | Logs to k8s|
+| | | App +--`
+| | +-----+               |          +-----------------------+
+| | +---------+   Publish to blobs   |                       |
+| `>| Fluentd +--------------------->|   Azure Blob Storage  |
```

Comment:
I'm not sure I understand the question you're posing here.

If the k8s cluster requires its own storage account to operate properly, then yes, I believe that should be separate from the more general "logs" storage account